### DISCUSSION TITLE
Deprecated and later removed NamespacedAttributeBag

### DISCUSSION THREAD

**Discussion by AndreRudolph** [symfony/symfony#45852] [Q&A]
I wanted to upgrade Symfony from 5.4.x to 6.x and noticed that the class Symfony\Component\HttpFoundation\Session\Attribute\NamespacedAttributeBag was deprecated in 5.3.x and apparently removed in 6.x. However, I can't find any documentation (neither in the code itself nor in the release notes) what the alternative is. Is there any? My code is relying on having such a bag that supports nested namespaces.

Thanks in advance for your help.

**Comment by MrYamous** [comment]
Hi, 

According to Symfony repo, this class seems to be [still present in 5.4](https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/HttpFoundation/Session/Attribute/NamespacedAttributeBag.php) and [removed in 6.0](https://github.com/symfony/symfony/tree/6.0/src/Symfony/Component/HttpFoundation/Session/Attribute).

Could you please check that your project has not been upgraded to Symfony 6.x ?

**Comment by AndreRudolph** [comment]
Hi,

sorry, I confused with the versions. I have already upgraded from 5.3.x to 5.4.x and wanted to upgrade to symfony 6. Thus I need to look for an alternative and was wondering why symfony deprecated and even removed them. Did they move it to another component or something else?

**Reply by fyrye** [reply]
It was removed outright in 6.x, suggesting to create your own class to provide the functionality as of 5.3+. 

> The NamespacedAttributeBag class is deprecated since Symfony 5.3. If you need this feature, you will have to implement the class yourself. [[sic]](https://symfony.com/doc/5.4/session.html)

As for the reasoning: see the ticket #32616 and associated pull-request #39855

In general, you can just copy the Symfony 5.3 `NamespaceAttributeBag` into your application code and wire it as a service. However, other changes to Symfony will require you to maintain the port.

**Comment by AndreRudolph** [comment]
Thank you very much for your help. This discussion can be closed then.
